packages/docs/docs/guides/5-request-client/2-erc20-payment-detection.ts

```diff
@@ -0,0 +1,211 @@
+(async () => {
+  const request = await requestNetwork.createRequest(addressBasedCreateParams);
+
+  console.log('Request created with erc20 address based payment network:');
+  console.log(request.requestId);
+
+  // The payer can add a refund address. For example when accepting the request
+  await request.accept(payerIdentity, {
+    refundAddress: '0xfb7914D846ab64A44c4e65dd38F4D004d29F911F',
+  });
+})();
+
+/**
+ * ### Paying an erc20 address based request
+ * 
+ * Requests created with the address based payment network must be manually paid by sending an erc20 transfer to the payment address.
+ * You can do this by calling `transfer(to, amount)` method of the erc20 token. `to` is the payment address and `amount` the amount the payer wants to pay.
+ */
+
+/**
+ * ## Request creation with the proxy contract payment network
+ *
+ * To create a request using erc20 proxy contract payment network, we need to provide the payment network parameters to the request creation parameters.
+ * Proxy contract payment network parameters are the same as the address based payment network: the payment and eventually the refund address.
+ */
+
+const proxyContractPaymentNetwork: RequestNetwork.Types.Payment.IPaymentNetworkCreateParameters = {
+  id: RequestNetwork.Types.Payment.PAYMENT_NETWORK_ID.ERC20_PROXY_CONTRACT,
+  parameters: {
+    // eslint-disable-next-line spellcheck/spell-checker
+    paymentAddress: '0x92FC764853A9A0287b7587E59aDa47165b3B2675',
+  },
+};
+
+const proxyContractCreateParams = {
+  proxyContractPaymentNetwork,
+  requestInfo,
+  signer: payeeIdentity,
+};
+
+// Finally create the request and print its id
+(async () => {
+  const request = await requestNetwork.createRequest(proxyContractCreateParams);
+
+  console.log(`Request created with erc20 proxy contract payment network: ${request.requestId}`);
+
+  // The payer can add a refund address. For example when accepting the request
+  await request.accept(payerIdentity, {
+    refundAddress: '0xfb7914D846ab64A44c4e65dd38F4D004d29F911F',
+  });
+})();
+
+// eslint-disable-next-line spellcheck/spell-checker
+/**
+ * ### Paying an erc20 proxy contract request manually
+ * 
+ * Payments for requests using ERC20 proxy contract payment network are documented in an Ethereum smart contract: *ERC20Proxy*. This smart contract is available on Ethereum [mainnet](https://etherscan.io/address/0x5f821c20947ff9be22e823edc5b3c709b33121b3) and Rinkeby [testnet](https://rinkeby.etherscan.io/address/0x162edb802fae75b9ee4288345735008ba51a4ec9).
+ * To perform a payment, the payer has first to allow the smart contract to transfer funds from his address. You must call `approve(spender, amount)` method of the specific ERC20 token smart contract. `spender` is the address of the proxy smart contract (`0x5f821c20947ff9be22e823edc5b3c709b33121b3` on mainnet, `0x162edb802fae75b9ee4288345735008ba51a4ec9` on Rinkeby), `amount` is the maximum amount you want to allow the proxy contract to transfer funds.
```

Comment:
maybe add something to explain why it's not a security issue to approve an erc20 proxy contract. This can be very scary

Reply:
Agreed, maybe we should even have a separate page in the docs explaining why we need to approve (it can be a lengthy explanation).